lib/internal/event_target.js

```diff
@@ -200,10 +200,12 @@ class Listener {
       previous.next = this;
     this.previous = previous;
     this.listener = listener;
+    // TODO(benjamingr) these 4 can be 'flags' to save 3 slots
```

Comment:
This is in order to fix an unrelated bug where we cache the next handler in our `dispatchEvent` and still call it. Added a test or this case below.